server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/filtering/RuleDTO.java

```diff
@@ -215,22 +217,45 @@ public static ImmutableList<RuleDTO> from(List<Rule> rules) {
     public static RuleDTO from(Rule rule) {
         return new RuleDTO(rule.getId().asString(),
                 rule.getName(),
-                ConditionDTO.from(rule.getCondition()),
+                rule.getConditions().stream().map(ConditionDTO::from).collect(Collectors.toList()),
+                rule.getConditionCombiner(),
                 ActionDTO.from(rule.getAction()));
     }
 
     private final String id;
     private final String name;
     private final ConditionDTO conditionDTO;
+    private final List<ConditionDTO> conditionDTOs;
+    private final Rule.ConditionCombiner conditionCombiner;
     private final ActionDTO actionDTO;
 
+    @JsonCreator
+    public RuleDTO(@JsonProperty("id") String id,
+                   @JsonProperty("name") String name,
+                   @JsonProperty("conditions") List<ConditionDTO> conditionDTOs,
+                   @JsonProperty("conditionCombiner") Rule.ConditionCombiner conditionCombiner,
+                   @JsonProperty("action") ActionDTO actionDTO) {
+        this.name = name;
+        this.conditionDTO = null;
+        this.conditionDTOs = conditionDTOs;
+        this.conditionCombiner = conditionCombiner;
+        this.actionDTO = actionDTO;
+        Preconditions.checkNotNull(id);
```

Comment:
preconditions should go first

Reply:
Please refactor the initial constructor as well that has preconditions at the end by mistake